functions/src/submitApplication/index.js

```diff
@@ -36,12 +36,13 @@ exports.submitApplicationHacker = functions.https.onRequest((request, response)
       // create a batch job
       const batch = db.batch();
 
-      // create a new document in hacker_quick_info and set its value
+      // create a new document in hacker_quick_info and hacker_main_info with the same ID
       const hackerQuickInfoRef = db.collection('hacker_quick_info').doc();
-      batch.set(hackerQuickInfoRef, hacker.hacker_quick_info);
+      const hackerId = hackerQuickInfoRef.id; // use same document ID for this hacker's documents
```

Comment:
What do you think about storing the id inside the doc as well? Might make things easier on the client js side

Reply:
That sounds like a good idea, I looked it up and [querying by document id seems kinda verbose](https://stackoverflow.com/a/48467056).

Reply:
On a similar note, should we store `hacker_quick_info` in `hacker_main_info` as well (we could call them `hacker_quick_info` and `hacker_full_info`)? i.e. `hacker_quick_info` would be sort of like a cache / index structure. 

The reason I mention this is because it would also make filtering easier. Ex: in our current set up filtering by `score > 8` and `travel_reimbursement == 2` would require us to check both documents since `score` would be in `hacker_quick_info`.

Reply:
@chamkank yeah I agree, `hacker_full_info` should contain everything, and `hacker_quick_info` would only contain a subset of it